### PR TITLE
fix quest demon souls

### PR DESCRIPTION
**Pull request by Username5173**
**Changes proposed:**

- If you use the quest item the npc spawns, gives credit for the killed demons and despawns their corpses

**Issues addressed:** Closes #  (insert issue tracker number)
fixes the quest demon souls
https://www.wowhead.com/quest=37653/demon-souls

**Tests performed:** (Does it build, tested in-game, etc.)
tested ingame

**Known issues and TODO list:** (add/remove lines as needed)

- On retail the npc uses a spell on the npcs instead of making them despawn

### CODE REVIEW

File: src/server/scripts/BrokenIsles/zone_azsuna.cpp [hunk 1]
```diff
@@ -87,8 +87,80 @@ class scene_azsuna_runes : public SceneScript
...
+    struct questnpc_soul_gemAI : public ScriptedAI
+    {
+        questnpc_soul_gemAI(Creature* creature) : ScriptedAI(creature) { }
+        bool check = false;
+
+        virtual void UpdateAI(uint32 diff) override
+        {
+            if(check == false)
+            { 
+            me->CastSpell(me, 186465);
```
Comment (Lartty96) [inline src/server/scripts/BrokenIsles/zone_azsuna.cpp]: Indent level missing

File: src/server/scripts/BrokenIsles/zone_azsuna.cpp [hunk 2]
```diff
@@ -87,8 +87,80 @@ class scene_azsuna_runes : public SceneScript
...
+        void checkfordeaddemons(Creature* creature)
+        {
+            std::list<Creature*> targets = creature->FindAllCreaturesInRange(15.0f);
+            Player* Owner = creature->GetOwner()->ToPlayer();
+
+            for (auto itr : targets)
+            {
+                if(itr->IsAlive() == false)
+                { 
+                switch (itr->GetEntry())
```
[the remark this answers is not in the document]
Comment (Username5173) [reply src/server/scripts/BrokenIsles/zone_azsuna.cpp]: Changed it. Thx for the reminder.

**Comment by Username5173** [comment]
Do you still have the sniffs from azsuna?

File: sql/updates/world/master/2018_08_18_98_world.sql [hunk 3]
```diff
@@ -0,0 +1,6 @@
+-- Demon Hunter SAI
```
Comment (Traesh) [inline sql/updates/world/master/2018_08_18_98_world.sql]: Can you explain me what's the point of this script ?

File: src/server/scripts/BrokenIsles/zone_azsuna.cpp [hunk 4]
```diff
@@ -87,8 +87,64 @@ class scene_azsuna_runes : public SceneScript
...
+    {
+        questnpc_soul_gemAI(Creature* creature) : ScriptedAI(creature) { }
+        bool check = false;
+
+        virtual void UpdateAI(uint32 diff) override
+        {
+            if(check == false)
+            { 
+            checkfordeaddemons(me);
+            check = true;
```
Comment (Traesh) [inline src/server/scripts/BrokenIsles/zone_azsuna.cpp]: If you want to do it only once, you should use the Reset() method and drop the "check" var :)

edit: and drop the "virtual" in front of the method, override after is enough

**Review by Traesh — CHANGES_REQUESTED** [review]

File: sql/updates/world/master/2018_08_18_98_world.sql (hunk 3 above)
Comment (Username5173) [reply sql/updates/world/master/2018_08_18_98_world.sql]: It fixes https://www.wowhead.com/quest=39877/in-the-loop

File: src/server/scripts/BrokenIsles/zone_azsuna.cpp (hunk 4 above)
Comment (Username5173) [reply src/server/scripts/BrokenIsles/zone_azsuna.cpp]: thx

**Review by Traesh — APPROVED** [review]

**Pull request merged**